manifests/init.pp

```diff
@@ -255,6 +255,7 @@
     django_secret_key              => $pulpcore_django_secret_key,
     content_service_worker_timeout => $pulpcore_content_service_worker_timeout,
     api_service_worker_timeout     => $pulpcore_api_service_worker_timeout,
+    api_client_auth_cn             => {"${foreman_host}" => 'admin'},
```

Comment:
```suggestion
    api_client_auth_cn             => {$foreman_host => 'admin'},
```

Reply:
Reading more, there's no guarantee that the CN on the foreman host is the same as the client cert. I'm trying to decouple that. For example, it may be `foreman.example.com` but the host uses `foreman01.example.com` as the hostname. I'm not entirely sure what the right approach is today, but perhaps the Foreman Proxy's trusted host settings is a better indicator. That typically includes it.

Reply:
The trusted hosts can be a list right? So are suggesting crafting a mapping of each trusted host entry to admin?

Reply:
I'm not entirely sure. That's what I also thought about. I think it makes sense to treat everything that can connect to a Smart Proxy with full permissions also as something that can connect to the Pulp instance with full permissions.